### ISSUE TITLE
epm-error on some smac-data

### ISSUE THREAD

**Issue by shukon** [automl/CAVE#215]
happens in fanova, ablation and sometimes in cost_over_time.

There are 26 parameters and 138 instance features in the example below.

```
ValueError                                Traceback (most recent call last)
<ipython-input-11-cb41bb471f19> in <module>
----> 1 cave.cave_ablation()

(...)

~/virtual_environments/ve_release/lib/python3.6/site-packages/pimp/evaluator/ablation.py in run(self)
    287 
    288         # Predict source and target performance to later use it to predict the %improvement a parameter causes
--> 289         source_mean, source_var = self._predict_over_instance_set(impute_inactive_values(self.source))
    290         prev_performance = source_mean
    291         target_mean, target_var = self._predict_over_instance_set(impute_inactive_values(self.target))

~/virtual_environments/ve_release/lib/python3.6/site-packages/pimp/evaluator/ablation.py in _predict_over_instance_set(self, config)
    387             the variance over the instance set. If logged values are used, the variance might not be able to be used
    388         """
--> 389         mean, var = self.model.predict_marginalized_over_instances(np.array([config.get_array()]))
    390         return mean, var
    391 

~/virtual_environments/ve_release/lib/python3.6/site-packages/smac/epm/base_epm.py in predict_marginalized_over_instances(self, X)
    240             raise ValueError('Expected 2d array, got %dd array!' % len(X.shape))
    241         if X.shape[1] != len(self.types):
--> 242             raise ValueError('Rows in X should have %d entries but have %d!' % (len(self.types), X.shape[1]))
    243 
    244         if self.instance_features is None or \

ValueError: Rows in X should have 164 entries but have 26!
```

**Comment by shukon** [comment]
Actually this seems to happen when including test-instances and might be related to changes in [SMAC's BaseEPM](https://github.com/automl/SMAC3/blob/master/smac/epm/base_epm.py) in 0.10.0.

**Comment by shukon** [comment]
@AndreBiedenkapp I think this is something that affects all kinds of calls to `
predict_marginalized_over_instances`, because the [new checks](https://github.com/automl/SMAC3/blob/b5c398655a5ff4e15ecfd3e23afda5303cdcbc06/smac/epm/base_epm.py#L241-L242) enforce that we specify the actual features over which to marginalize... while everywhere in pimp, we just [throw in the configs](https://github.com/automl/ParameterImportance/blob/ca347be8d9249d98633b073ea0eba12742dc51f2/pimp/importance/importance.py#L131-L135)... is this a design issue I'm not aware of? Or did we just use the epm in way that was not intended (maybe @mfeurer)?

**Comment by shukon** [comment]
This should be solved with CAVE 1.1.8 and PIMP 1.0.6
